src/test/java/seedu/address/logic/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,126 @@
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        ViewCommand viewCommand = new ViewCommand(outOfBoundIndex);
+
+        assertCommandFailure(viewCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredList_throwsCommandException() {
+        showPersonAtIndex(model, INDEX_FIRST_PERSON);
+
+        Index outOfBoundIndex = INDEX_SECOND_PERSON;
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getPersonList().size());
+
+        ViewCommand viewCommand = new ViewCommand(outOfBoundIndex);
+
+        assertCommandFailure(viewCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void equals() {
+        ViewCommand viewFirstCommand = new ViewCommand(INDEX_FIRST_PERSON);
+        ViewCommand viewSecondCommand = new ViewCommand(INDEX_SECOND_PERSON);
+
+        // same object -> returns true
+        assertEquals(viewFirstCommand, viewFirstCommand);
+
+        // same values -> returns true
+        ViewCommand viewFirstCommandCopy = new ViewCommand(INDEX_FIRST_PERSON);
+        assertEquals(viewFirstCommand, viewFirstCommandCopy);
+
+        // different types -> returns false
+        assertNotEquals(1, viewFirstCommand);
+
+        // null -> returns false
+        assertNotEquals(null, viewFirstCommand);
+
+        // different person -> returns false
+        assertNotEquals(viewFirstCommand, viewSecondCommand);
+    }
+
+    @Test
+    public void execute_emptyAddressBook_throwsCommandException() {
+        // Empty the address book
+        model = new ModelManager();
+        ViewCommand viewCommand = new ViewCommand(Index.fromOneBased(1));
+
+        assertCommandFailure(viewCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void equals_otherObjectNotInstanceOfViewCommand_returnsFalse() {
+        // Create a ViewCommand object
+        ViewCommand viewCommand = new ViewCommand(Index.fromOneBased(1));
+
+        // Create another object of a different class
+        Object otherObject = new Object();
+
+        // Check if they are equal
+        assertFalse(viewCommand.equals(otherObject));
+    }
+
+    @Test
+    public void toStringMethod() {
+        Index targetIndex = Index.fromOneBased(1);
+        ViewCommand viewCommand = new ViewCommand(targetIndex);
+        String expected = ViewCommand.class.getCanonicalName() + "{targetIndex=" + targetIndex + "}";
+        assertEquals(expected, viewCommand.toString());
+    }
+
+    /**
+     * Updates {@code model}'s filtered list to show no one.
+     */
+    private void showNoPerson(Model model) {
+        model.updateFilteredPersonList(p -> false);
+
+        assertTrue(model.getFilteredPersonList().isEmpty());
+    }
```

Comment:
Is this test relevant to View class? Since its not calling any view related methods, nor is it used anywhere. Its also not marked with `@Test`